src/main/java/org/myrobotlab/service/InMoov.java

```diff
@@ -342,21 +384,47 @@ public void loadGestures() {
   }
 
   /**
-   * This method will look at all of the .py files in a directory. One by one it
+   * This method will try to launch a python command
+   * with error handling
+   */
+  public String execGesture(String gesture) {
+    lastGestureExecuted = gesture;
+    String ret;
+    if (python == null) {
+      log.warn("execGesture : No jython engine...");
+      return null;
+    }
+    subscribe(python.getName(), "publishStatus", this.getName(), "onGestureStatus");
+
+    startedGesture(lastGestureExecuted);
+    ret = Utils.execPy(gesture);
+    return ret;
+  }
+
+  public void onGestureStatus(Status status) {
+    if (!(status == Status.success()) && !(status == Status.warn("Python process killed !"))) {
+      speakAlert(languagePack.get("GESTURE_ERROR"));
+    }
+    finishedGesture(lastGestureExecuted);
+    unsubscribe(python.getName(), "publishStatus", this.getName(), "onGestureStatus");
+  }
+
+  /**
+   * This blocking method will look at all of the .py files in a directory. One by one it
    * will load the files into the python interpreter. A gesture python file
    * should contain 1 method definition that is the same as the filename.
    * 
    * @param directory
    *          - the directory that contains the gesture python files.
    */
-  public void loadGestures(String directory) {
+  public Boolean loadGestures(String directory) {
```

Comment:
Should return a native "boolean"  not a Boolean here.  Booleans can be null.  This method,  if  it returns anything should only be true or false.  not "null" as is possible with a Boolean as a return type.